dvc/remote/base.py

```diff
@@ -667,16 +667,29 @@ def _checkout_file(
         self, path_info, checksum, force, progress_callback=None
     ):
         cache_info = self.checksum_to_path_info(checksum)
-        if self.exists(path_info):
-            msg = "data '{}' exists. Removing before checkout."
-            logger.warning(msg.format(str(path_info)))
-            self.safe_remove(path_info, force=force)
 
-        self.link(cache_info, path_info)
-        self.state.save_link(path_info)
-        self.state.save(path_info, checksum)
-        if progress_callback:
-            progress_callback.update(str(path_info))
+        if self._should_link_from_cache(path_info, checksum, force):
+            self.link(cache_info, path_info)
+            self.state.save_link(path_info)
+            self.state.save(path_info, checksum)
+            if progress_callback:
+                progress_callback.update(str(path_info))
+
+    def _should_link_from_cache(self, path_info, checksum, force):
+        if self.exists(path_info):
+            if (
+                    self.state.get(path_info) == checksum
+                    and not force
+                    # NOTE In case of protected repo, we need to overwrite even
+                    # identical file to make it protected again
+                    and not self.protected
```

Comment:
So, it means that if the repo is protected then we basically keep the existing behavior - we'll write warnings for 

dvc run -o metrics.txt "echo 0.9643 > metrics.txt" 

every time you run it?

Reply:
Yes, in case of protected cache existing behaviour will remain.